src/hotspot/os/linux/os_linux.cpp

```diff
@@ -1825,10 +1843,11 @@ void os::print_dll_info(outputStream *st) {
   pid_t pid = os::Linux::gettid();
 
   jio_snprintf(fname, sizeof(fname), "/proc/%d/maps", pid);
-
-  if (!_print_ascii_file(fname, st)) {
+  unsigned num = 0;
+  if (!_print_ascii_file(fname, st, &num)) {
     st->print_cr("Can not get library information for pid = %d", pid);
   }
+  st->print_cr("Total number of mappings: %u", num);
```

Comment:
Should we only print the number of mappings if  `_print_ascii_file` returns `true`? I.e. to avoid the hypothetical output of:

```
Can not get library information for pid = <pid>
Total number of mappings: 0
```

How about?

```
 unsigned num = 0;
  if (!_print_ascii_file(fname, st, &num)) {
    st->print_cr("Can not get library information for pid = %d", pid);
  } else {
    st->print_cr("Total number of mappings: %u", num);
  }
```

Reply:
Good catch. Yes, that makes sense.